File: src/repositories/interfaces/gym.repository.ts
import { Failure } from "../../common/types";
import { IGymEntity, IUserEntity, LocationEntity } from "../../entities";

export interface IGymRepository {
  create(gym: Omit<IGymEntity, "_id">): Promise<IGymEntity>;

  getNearest(
    location: LocationEntity,
    meters: number,
    amount?: number
  ): Promise<IGymEntity[]>;

  getByUser(user: IUserEntity): Promise<IGymEntity | Failure>;

  get(gymId: string): Promise<IGymEntity | Failure>;

  deleteAll(): Promise<void>;
}

File: lib/src/repositories/interfaces/gym.repository.dart
import 'package:dartz/dartz.dart';

import '../../entities/entities.dart';

abstract class IGymRepository {
  Future<Either<List<GymEntity>, Failure>> getNearestGyms({
    required LocationEntity location,
    required double distance,
    int? amount,
  });

  Future<Either<GymEntity, Failure>> getGym(String accessToken);
}
